File: C++/프로그래머스/0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<int>> lines) {
    map<int, int> line_list;
    
    for(const auto& l: lines) {
        for (int i = l[0]; i < l[1]; i++) {
            line_list[i]++;
        }
    }
    
    int answer = 0;
    for (const auto& l: line_list)
        if (l.second > 1) answer++;
    
    return answer;
}

File: Go/프로그래머스/0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.go
func solution(lines [][]int) int {
    line_map := make(map[int]int)
    
    for _, l := range lines {
        for i := l[0]; i < l[1]; i++ {
            line_map[i]++
        }
    }
    
    answer := 0
    for _, v := range line_map {
        if v > 1 {
            answer++
        }
    }
    return answer
}